src/lib.rs

```diff
@@ -36,6 +36,63 @@ pub use types::geometrycollection::GeometryCollection;
 
 pub use types::feature::Feature;
 pub use types::featurecollection::FeatureCollection;
-pub use types::featurecollection::from_str;
+
+macro_rules! expect_string {
+    ($value:expr) => (try!(
+        match $value.as_string() {
+            Some(v) => Ok(v),
+            None => Err({use GeoJsonError; GeoJsonError::new("Expected string value")})
+        }
+    ))
+}
+
+macro_rules! expect_f64 {
+    ($value:expr) => (try!(
+        match $value.as_f64() {
+            Some(v) => Ok(v),
+            None => Err({use GeoJsonError; GeoJsonError::new("Expected f64 value")})
+        }
+    ))
+}
+
+macro_rules! expect_array {
+    ($value:expr) => (try!(
+        match $value.as_array() {
+            Some(v) => Ok(v),
+            None => Err({use GeoJsonError; GeoJsonError::new("Expected array value")})
+        }
+    ))
+}
+
+macro_rules! expect_object {
+    ($value:expr) => (try!(
+        match $value.as_object() {
+            Some(v) => Ok(v),
+            None => Err({use GeoJsonError; GeoJsonError::new("Expected object value")})
+        }
+    ))
+}
+
+macro_rules! expect_property {
+    ($obj:expr, $name:expr, $desc:expr) => (
+        match $obj.get($name) {
+            Some(v) => v,
+            None => return Err({use GeoJsonError; GeoJsonError::new($desc)}),
+        };
+    )
+}
 
 mod types;
+
+#[derive(Copy)]
+pub struct GeoJsonError {
+    pub desc: &'static str,
+}
+
+impl GeoJsonError {
+    pub fn new(desc: &'static str) -> GeoJsonError {
+        return GeoJsonError{desc: desc};
+    }
+}
+
+pub type GeoJsonResult<T> = Result<T, GeoJsonError>;
```

Comment:
I'm following the example of `IoError`, returning an object that can encode more information than just a message, maybe the line and column number in the original JSON document, if that's available.